.circleci/config.yml

```diff
@@ -138,10 +138,18 @@ jobs:
     docker: *docker-custom-nodejs
     environment:
       XOD_OUTPUT: /tmp/tabtests-cpp
+      XOD_ARDUINO_CLI: /tmp/arduino-cli/arduino-cli
     steps:
       - checkout
       - restore_cache: *restore-node_modules
       - run: *step-install
+      - run: *step-install-arduino-cli-on-linux
+      - run:
+          name: Install arduino:avr platform for arduino-cli
+          command: $XOD_ARDUINO_CLI core install arduino:avr
```

Comment:
Worth caching? In order to create less load on the arduino mirrors.

Reply:
@evgenykochetkov CircleCI cache is immutable, so it can't be updated with the same key.

Proposed solution (not tested):
```yaml
- restore_cache:
    name: Restore arduino-cli cache
    keys:
      - arduino15-v1-{{ arch }}-{{ .Branch }}-
      - arduino15-v1-{{ arch }}-
# Install arduino:avr platform for arduino-cli
# Test AVR Size
- save_cache:
    name: Save arduino-cli cache
    # update cache only if package_index.json changed
    key: arduino15-v1-{{ arch }}-{{ .Branch }}-{{ checksum "~/.arduino15/package_index.json" }}
    paths:
      - ~/.arduino15/staging # only archives
```